libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -1394,7 +1394,8 @@ void RCOutput::dshot_send(pwm_group &group, uint32_t time_out_us)
 
     // if we are sharing UP channels then it might have taken a long time to get here,
     // if there's not enough time to actually send a pulse then cancel
-    if (AP_HAL::micros() + group.dshot_pulse_time_us > time_out_us) {
+
+    if (AP_HAL::micros64() + group.dshot_pulse_time_us > time_out_us) {
```

Comment:
Was this line the actual bug?

Reply:
We don't know. I'm currently working my way through a "proper" fix to try and figure it out

Reply:
There are a few places that could be "the" bug. Unfortunately what we have reproduce is not necessarily in the same place that the bug was seen.